tests/test_caching.py

```diff
@@ -0,0 +1,160 @@
+import logging
+
+import pandas as pd
+import pytest
+
+from hamilton import base
+from hamilton.driver import Driver
+from hamilton.experimental.h_cache import (
+    CachingAdapter,
+    read_feather,
+    read_json,
+    read_parquet,
+    write_feather,
+    write_json,
+    write_parquet,
+)
+from tests import nodes
+
+
+@pytest.fixture
+def df():
+    return pd.DataFrame(
+        {
+            "a": [1, 2, 3],
+            "b": [4, 5, 6],
+        }
+    )
+
+
+def _read_write_df_test(df, filepath, read_fn, write_fn):
+    write_fn(df, filepath)
+    pd.testing.assert_frame_equal(read_fn(filepath), df)
+
+
+def test_feather(df, tmp_path):
```

Comment:
is tmp_path a pytest thing? I don't see a fixture.

Reply:
Yes, it's a built-in pytest fixture: https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-fixture

Reply:
TIL!